app/controllers/secrets_controller.rb

```diff
@@ -54,6 +54,26 @@ def duplicates
       sort_by { |_, v| -v.size }
   end
 
+  def resolve
+    params.require([:project_id, :deploy_group, :keys])
+
+    project = Project.find(params[:project_id])
```

Comment:
kinda strange that deploy group is by permalink and project is by id ... most other controller do project by permalink

Reply:
yep, i know -- i couldn't figure out how to wire it up, i can change this to support both if you want -- i was looking at what we currently do in scooter, and the new mutating webhook service, and they're both looking up projects by id, because they look up the project first.

What would it look like to support both? I didn't get how that was wired in the current state.

Reply:
supporting both is fine too ... I think we even do that somewhere already ... `id.match?(/^\d+$/) ? find_by_id : find_by_permalink`